### DISCUSSION TITLE
Controlling AIT for specific regions in domain

### DISCUSSION THREAD

**Discussion by Vojta-Salek** [firemodels/fds#13410] [Q&A] [answered]
Hello FDS community,

Do you know of a way to easily assign multiple auto ignition temperature (AIT) values for specific regions of the domain? For example, I would like to prevent combustion in certain parts of the domain and prescribe specific AIT values elsewhere, assuming a single fuel. Another case would be having different AITs in regions with large computational cells, where AIT needs to be lowered compared to fine mesh regions.

I am aware of the `AIT_EXCLUSION_ZONE` feature, which sets AIT = 0 K for the excluded zone. However, I am looking for the opposite behavior - assigning high AIT (e.g., 2000 °C) for the excluded zone to effectively prevent combustion. 

I am also aware about the possibility to use control devices for spark ignition based on temperature, but I’d prefer to avoid tracking the temperature of each cell in a potentially large regions and introducing a spark only when a certain cell temperature reaches the desired AIT.

Ideally, it would be nice to simply add something like `AIT_EXCLUSION_ZONE_TEMPERATURE(N)` to specify AIT for the region defined by `AIT_EXCLUSION_ZONE(N)`. I believe this could provide more control over AIT in your domain. I will ask for this feature properly using the requests if this discussions does not reveal any existing solution I may have overlooked.

Thank you in advance for any ideas!

Vojtech Salek

**Comment by rmcdermo** [comment]
I'm looking into this.

**Comment by rmcdermo** [comment, marked as answer]
OK, I've added the code you suggested ( see PR #13413 ).  You can either compile yourself, or wait for a nightly build to post.  Here is a short test case based on the `fds/Verification/Fires/simple_test.fds` case.  Here the AIT_EXCLUSION_ZONE is added to the right (+y) half of the domain.

```
&HEAD CHID='ait_per_zone', TITLE='Simple demonstration case.'  /

&MESH IJK=36,24,24, XB=0.0,3.6,0.0,2.4,0.0,2.4 / 

&TIME T_END=60.0 /

&DUMP DT_SL3D=10 /

&REAC FUEL='PROPANE', SOOT_YIELD=0.01
      AIT_EXCLUSION_ZONE=0,3.6,1.2,2.4,0,2.4
      AIT_EXCLUSION_ZONE_TEMPERATURE=3000 /

&SURF ID='BURNER', HRRPUA=1000., COLOR='RED' /

&OBST XB=0.0,0.4,1.0,1.4,0.0,0.2 /
&VENT XB=0.0,0.4,1.0,1.4,0.2,0.2, SURF_ID='BURNER' /

&VENT XB=3.6,3.6,0.8,1.6,0.0,2.0, SURF_ID='OPEN' /

&SLCF PBY=1.2, QUANTITY='TEMPERATURE', VECTOR=T /
&SLCF PBX=0.2, QUANTITY='HRRPUV', CELL_CENTERED=T /

&SLCF XB=0.0,3.6,0.0,2.4,0.0,2.4, QUANTITY='HRRPUV', CELL_CENTERED=T /
&SLCF XB=0.0,3.6,0.0,2.4,0.0,2.4, QUANTITY='MASS FRACTION', SPEC_ID='PROPANE' CELL_CENTERED=T /
&SLCF XB=0.0,3.6,0.0,2.4,0.0,2.4, QUANTITY='MASS FRACTION', SPEC_ID='OXYGEN' CELL_CENTERED=T /

&TAIL /
```

![Screenshot 2024-09-09 at 11 58 12 AM](https://github.com/user-attachments/assets/17950fb7-3c34-42a9-9a47-d4dae8d65355)

**Reply by Vojta-Salek** [reply]
Thank you very much for addressing this idea, I very much appreciate the continuous support you provide.

I'll do some testing in my code and will let you know in case I encounter any unexpected behavior.

**Reply by Vojta-Salek** [reply]
Finally, I had time to continue with some other tests. Am I right that the number of excluded zones is limited to 10? Is there a way to increase this limit? I know it may sound silly, but I think I've found a way to meaningfully use more than 10 zones for a single reaction.

Thank you for you support!

**Reply by mcgratta** [reply]
If you are compiling yourself, this parameter is set in `prec.f90', `MAX_AIT_EXCLUSION_ZONES`.
